radix-engine-interface/src/model/resource/non_fungible_local_id.rs

```diff
@@ -238,32 +254,38 @@ impl FromStr for NonFungibleLocalId {
         let local_id = if s.starts_with("<") && s.ends_with(">") {
             NonFungibleLocalId::String(s[1..s.len() - 1].to_string())
         } else if s.starts_with("#") && s.ends_with("#") {
-            NonFungibleLocalId::Number(
+            NonFungibleLocalId::Integer(
                 u64::from_str_radix(&s[1..s.len() - 1], 10)
-                    .map_err(|_| ParseNonFungibleLocalIdError::InvalidNumber)?,
+                    .map_err(|_| ParseNonFungibleLocalIdError::InvalidInteger)?,
             )
         } else if s.starts_with("[") && s.ends_with("]") {
             NonFungibleLocalId::Bytes(
                 hex::decode(&s[1..s.len() - 1])
                     .map_err(|_| ParseNonFungibleLocalIdError::InvalidBytes)?,
             )
         } else if s.starts_with("{") && s.ends_with("}") {
-            let hex: String = s[1..s.len() - 1]
-                .chars()
-                .into_iter()
-                .filter(|c| *c != '-')
-                .collect();
-            NonFungibleLocalId::UUID(
-                u128::from_str_radix(&hex, 16)
-                    .map_err(|_| ParseNonFungibleLocalIdError::InvalidUUID)?,
-            )
+            let chars: Vec<char> = s[1..s.len() - 1].chars().collect();
+            if chars.len() == 32 + 4
+                && chars[8] == '-'
+                && chars[13] == '-'
+                && chars[18] == '-'
+                && chars[23] == '-'
+            {
+                let hyphen_stripped: String = chars.into_iter().filter(|c| *c != '-').collect();
+                NonFungibleLocalId::UUID(
```

Comment:
I think we need to check that hyphen_stripped.len() equals 32, otherwise eg `{----------..--0}` would be valid I think?

Reply:
`u128::from_str_radix` would catch that.  I will double check as well.